src/home/homed-home-bus.c

```diff
@@ -436,10 +436,32 @@ int bus_home_update_record(
         if ((flags & ~SD_HOMED_UPDATE_FLAGS_ALL) != 0)
                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid flags provided.");
 
+        if (blobs) {
+                const char *failed = NULL;
+                r = user_record_ensure_blob_manifest(hr, blobs, &failed);
+                if (r == -EINVAL)
+                        return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Provided blob files do not correspond to blob manifest.");
+                if (r < 0)
+                        return sd_bus_error_set_errnof(error, r, "Failed to generate hash for blob %s: %m", strnull(failed));
+        }
+
+        safe = user_record_self_changes_allowed(h->record, hr);
+        if (safe < 0) {
+                log_warning_errno(safe, "Failed to determine if changes to user record are safe, assuming not: %m");
+                safe = false;
+        } else if (safe) {
+                safe = bus_home_client_is_trusted(h, message);
```

Comment:
this of course is slightly skewed here. Because if the client is root this will cause us to use the org.freedesktop.home1.update-home-by-owner PK action, which is of course strictly speaking weird because root is not really the "owner" of such accounts...

hence, either fix this (i.e. add a "bool strict" param to bus_home_client_is_trusted() or so? or at least add a comment explaining why the sloppiness here doesn't matter too much